bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -186,6 +212,91 @@ private String getTransactionLogDirectory() {
         }
     }
 
+    /**
+     * This method gets the user specified config for ballerina recovery log name.
+     *
+     * @return string recovery log file name
+     */
+    private String getRecoveryLogBaseName() {
+        VariableKey recoveryLogNameKey =
+                new VariableKey(TRANSACTION_PACKAGE_ID, "recoveryLogName", PredefinedTypes.TYPE_STRING, false);
+        if (!ConfigMap.containsKey(recoveryLogNameKey)) {
+            return "recoveryLog";
+        } else {
+            return ((BString) ConfigMap.get(recoveryLogNameKey)).getValue();
+        }
+    }
+
+    /**
+     * This method gets the user specified config for ballerina recovery log directory.
+     *
+     * @return string recovery log directory
+     */
+    private Path getRecoveryLogDir() {
+        final Path projectRoot = Paths.get(RuntimeUtils.USER_DIR);
+        VariableKey recoveryLogDirKey =
+                new VariableKey(TRANSACTION_PACKAGE_ID, "recoveryLogDir", PredefinedTypes.TYPE_STRING, false);
+        if (!ConfigMap.containsKey(recoveryLogDirKey)) {
+            return projectRoot;
+        } else {
+            String logDir = ((BString) ConfigMap.get(recoveryLogDirKey)).getValue();
+            Path logDirPath = Paths.get(logDir);
+            Path recoveryLogDirectory;
+            if (!logDirPath.isAbsolute()) {
+                logDir = projectRoot.toAbsolutePath().toString() + File.separatorChar + logDir;
+                recoveryLogDirectory = Paths.get(logDir);
+            } else {
+                recoveryLogDirectory = logDirPath;
+            }
+            return recoveryLogDirectory;
```

Comment:
Return each element directly in the place

Reply:
fixed in [d629dbd](https://github.com/ballerina-platform/ballerina-lang/pull/42080/commits/d629dbd9b2347a153baf5317799ae05805f7b53c)